### ISSUE TITLE
cPanel apache error log location

### ISSUE THREAD

**Issue by tripflex** [richardforth/apache2buddy#302]
On cPanel servers, it seems that the `error_log` file is no longer stored in `/var/log/httpd/error_log` but stored in `/var/log/apache2/error_log` even though service is located at `/usr/sbin/httpd`

I just created a symlink for the time being, but may be good to update to check for error log in that location as well?

```
[ -- ] Distro: CentOS Linux
[ -- ] Version: 7.7.1908
[ -- ] cPanel Version: 11.82.0.16
[ -- ] The process listening on port 80 is /usr/sbin/httpd
[ -- ] The process running on port 80 is Apache/2.4.41 (cPanel).
[ -- ] The full path to the Apache config file is: /etc/apache2/conf/httpd.conf
[ >> ] Non-standard apache set-up, apache is running as UID: 99 (nobody). Expecting UID 48 ('apache' or 'www-data').
```

**Comment by richardforth** [comment]
Thats a real pain in the neck.
Not sure how I can fix that,
I hate all the inconsistency. Glad you managed to work around it for now.
Will leave this open but sot sure when I will get a chance to work on it.

**Comment by richardforth** [comment]
I came back to revisit this, but ultimately I may have to decide to close it off:

``` Non-standard apache set-up,```

This probably tells us all we need to know, bear in mind its a cPanel server and cPanel has its own rules about where it puts things. Apache2buddy wasnt really designed to work with nonstandard setups (though I do try to be accommodating where I can).

I got as far as this - but it always produces a file not found error for one or the other:

```bash
# grep -i reached /var/log/{apache2,httpd}/error.log
/var/log/apache2/error.log:[Sat Dec 28 05:15:08.165058 2019] [mpm_prefork:error] [pid 94341] AH00161: server reached MaxRequestWorkers setting, consider raising the MaxRequestWorkers setting
grep: /var/log/httpd/error.log: No such file or directory
```

basically `{apache2,httpd}` is a shell expansion that creates two paths:

/var/log/apache2/error.log
/var/log/httpd/error.log

What I really want is a simpler OR not AND, I tried pipe (|) but that gets interpreted by bash  and breaks the command. 

I'll keep stabbing at it to see if I can get a fix in but I want to avoid any unnecessary errors to console  to keep the output polished.

**Comment by richardforth** [comment]
Just a quick note to say I havent forgotten about this

**Comment by richardforth** [comment]
Since most of the log file location code is pure guesswork anyway (largely thanks to apache being inconsistent across platforms "and" not using absolute paths in its config files)...

I may change the way it checks this by producing a list of possible locations and having it loop through each list item checking if the file exists. That may be one easy way to solve this going forward...

Log file not found? Add it to the list in a PR.

I think this is ultimately the way forward.

**Comment by richardforth** [comment]
Was referenced in error by typo should have been 309, reopenning #302

**Comment by richardforth** [comment]
```
2195         # Check 17d : Large Logs in /var/log
2196         systemcheck_large_logs("/var/log/httpd");
2197         systemcheck_large_logs("/var/log/httpd24");
2198         systemcheck_large_logs("/var/log/apache2");
2199         systemcheck_large_logs("/var/log/php-fpm");
2200         systemcheck_large_logs("/usr/local/apache/logs");
2201         systemcheck_large_logs("/usr/local/apache2/logs");
2202         systemcheck_large_logs("/usr/local/httpd/logs");
```


---

```
 506 sub systemcheck_large_logs {
 507         my ($logdir) = @_;
 508         if ( -d $logdir ) {
 509                 my @logs;
 510                 my $logfiles_raw = find(sub {push @logs, $File::Find::name  if -s >= 1024000000;},  $logdir);
 511                 foreach my $log (@logs) {
 512                         chomp($log);
 513                         # Issue 255 skip reporting already gzipped logs, as they have already been rotated
 514                         if ( ! $log =~ m/\.gz$/ ) {
 515                                 my $size = -s $log;
 516                                 my $humansize = sprintf "%.2f", $size/1024/1024/1024;
 517                                 show_crit_box(); print $log . " --> " . $humansize . "GB\b\n";
 518                         }
 519                 }
 520                 if (@logs == 0) {
 521                         if ( ! $NOOK ) { show_ok_box(); print "${GREEN}No large log files were found in ${CYAN}$logdir${ENDC}.\n"; }
 522                 } else {
 523                         show_advisory_box(); print "${YELLOW}Consider setting up a log rotation policy.${ENDC}\n";
 524                         show_advisory_box(); print "${YELLOW}Note: Log rotation should already be set up under normal circumstances, so very${ENDC}\n";
 525                         show_advisory_box(); print "${YELLOW}Large error logs can indicate a fundmental issue with the website / web application.${ENDC}\n";
 526                 }
 527         }
 528         # silently proceed if the folder doesnt exist
 529 }
```

This feels like it should be OK....

**Comment by richardforth** [comment]
I think we are already checking that location

```
2198         systemcheck_large_logs("/var/log/apache2");
```

Closing, but if you think it needs more attention, can you clarify what is happening if you dont symlink it?
